tests/Platforms/AbstractMySQLPlatformTestCase.php

```diff
@@ -122,17 +122,17 @@ public function testDoesSupportSavePoints(): void
 
     public function getGenerateIndexSql(): string
     {
-        return 'CREATE INDEX my_idx ON mytable (user_name, last_login)';
+        return 'CREATE INDEX `my_idx` ON mytable (`user_name`, `last_login`)';
     }
 
     public function getGenerateUniqueIndexSql(): string
     {
-        return 'CREATE UNIQUE INDEX index_name ON test (test, test2)';
+        return 'CREATE UNIQUE INDEX `index_name` ON test (`test`, `test2`)';
     }
 
     protected function getGenerateForeignKeySql(): string
     {
-        return 'ALTER TABLE test ADD FOREIGN KEY (fk_name_id) REFERENCES other_table (id)';
+        return 'ALTER TABLE test ADD FOREIGN KEY (`fk_name_id`) REFERENCES `other_table` (`id`)';
```

Comment:
The table name here is still unquoted because `AbstractPlatform::getCreateForeignKeySQL()` accepts the table name as SQL, not as a literal. Consistent quoting can be achieved by introducing values objects to represent object names (https://github.com/doctrine/dbal/issues/4772).